web/html/src/manager/virtualization/GuestProperties.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+
+const React = require("react");
+const ReactDOM = require("react-dom");
+const {SmallPanel} = require("../../components/panel");
+const {Form, FormGroup, Text, Select} = require("../../components/input");
+const {SubmitButton, Button} = require("../../components/buttons");
+const Validation = require("../../components/validation");
+const {Messages} = require("../../components/messages");
+const MessagesUtils = require("../../components/messages").Utils;
+const PropTypes = React.PropTypes;
+
+/*
+ * Component editing a virtual machine properties
+ *
+ * Properties:
+ *   + host
+ *   + submitText
+ *   + submit: a function that takes one parameter with the form model, and a setMessages function
+ *   + initModel: a function that returns the default values for the model
+ */
+class GuestProperties extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    (this:any).onValidate = this.onValidate.bind(this);
```

Comment:
I think there is something I didn't get correctly since this is getting me a syntax error right before the '(':

```javascript
    clearFields = () = {
      this.setState({
        model: this.props.getInitialModel()
      });
    }
```

Reply:
done